IndomitableIbises/Bangazon-API-Sprint-1
# Training Programs and Employee Training Models/View/URL/Serializer/Faker

## Link to Ticket
Closes #9 

## Description of Proposed Changes

- Created Model/View/Url/Serializer/Faker Data for Training Programs and Employee Programmings
- Added comments and authorship
- Added Jacob's changes per his request for Orders
- Added Faker code for Department and Computer
- SH script is updated and works as intended
- Removed supervisor id on department per Steve's recommendation (eliminates circular dependency)

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout RCR-Training
```
1. code .
2. **Delete** existing db.sqlite3 file **and** make sure your server isn't running
3. Run ./django_data.sh API faker_factory for **PC**
or
django_data.sh API faker_factory for **MAC**
4. Look over **Training_Prog** and **Emp_Training** Model, View, Url, Serializer -- let me know if you have any questions
5. python manage.py runserver
6. You should see clickable urls on our page. Click on "http://localhost:8000/training/" and make sure you are able to add a training program (name and max attendee) as well as update a training program.
7. You should see clickable urls on our page. Click on "http://localhost:8000/employee_training/" and make sure you are able to see all foreign keys and select by id in the drop down fields
8.  All foreign keys on tables will display relevant name/title for that FK
9.  Removed duplicate imports and separated all files into Employee and Customer `sections`

## Impacted Areas in Application

List general components of the application that this PR will affect:

*models.py
*views.py
*serializers.py
*url.py
*faker_factory.py


## Mentions @username
@CashewRose @sirwin4 @jcarter0149 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
